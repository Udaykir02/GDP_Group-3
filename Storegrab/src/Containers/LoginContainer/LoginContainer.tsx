import { loginRequest } from '../../../actions/userActions';
import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import { useEffect, useState } from 'react';
import { TouchableOpacity, View, Text, TextInput, Button, Image, TouchableHighlight, Alert } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import createToBeImplementedStyle from "./LoginContainerStyle";


const LoginContainer = ({ navigation }: any) => {
    const styles = createToBeImplementedStyle();
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const dispatch = useDispatch();
    const { error } = useSelector((state: any)=> state.auth)
    const handleLogin = async () => {
        await dispatch(loginRequest(email, password));
        
    };
    const handleResetPassword = () => {
        navigation.navigate('Reset')
    }

    useEffect(()=>{
        // Display alert when error occurs
        if (error) {
            console.log(JSON.stringify(error))
            Alert.alert('Error', error, [{ text: 'OK' }]);
        }
    },[error])

    return (
        <View style={styles.container}>
            <View style={styles.imageContainer}>
                <Image
                    source={require('../../Assets/icons/bearcat_plaw.png')}
                    style={styles.image}
                />
            </View>
            <View style={styles.formContainer}>
                <TextInput
                    style={styles.input}
                    placeholder="Email"
                    value={email}
                    onChangeText={setEmail}
                    autoCapitalize={'none'}
                    testID='email-input'
                />
                <TextInput
                    style={styles.input}
                    placeholder="Password"
                    value={password}
                    onChangeText={setPassword}
                    secureTextEntry
                    autoCapitalize={'none'}
                    testID='password-input'
                />
                <TouchableHighlight
                    style={styles.button}
                    onPress={handleLogin}
                    underlayColor='#fff' testID='login-button'>
                    <Text style={[{fontSize: 15},styles.loginText]}>Login</Text>
                </TouchableHighlight>
                <Button title="Reset Password" onPress={handleResetPassword} />
            </View>

        </View>
    );
};

export default LoginContainer;
