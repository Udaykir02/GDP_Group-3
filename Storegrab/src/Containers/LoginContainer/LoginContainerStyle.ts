import { StyleSheet } from 'react-native';
export default function <C>() {
    return StyleSheet.create({
        container: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            paddingHorizontal: 20,
          },
          title: {
            fontSize: 24,
            fontWeight: 'bold',
            marginBottom: 20,
          },
          input: {
            width: '100%',
            height: 40,
            borderColor: 'gray',
            borderWidth: 1,
            borderRadius: 5,
            paddingHorizontal: 10,
            marginBottom: 10,
          },
          button: {
            backgroundColor: 'blue',
            width: '50%',
            height: 40,
            borderRadius: 5,
            justifyContent: 'center',
            alignItems: 'center',
            marginTop: 20,
          },
          buttonText: {
            color: 'white',
            fontSize: 16,
            fontWeight: 'bold',
          },
    })
}