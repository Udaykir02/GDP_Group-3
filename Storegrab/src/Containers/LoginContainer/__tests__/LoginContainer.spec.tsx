import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LoginContainer from '../LoginContainer';
import { toBeInTheDocument } from '@testing-library/jest-dom/matchers';

describe('LoginContainer', () => {
  it('renders without crashing', () => {
    const { container } = render(<LoginContainer />);
    expect(container).toBeTruthy();
  });

it('displays an error message when username is empty', async () => {
    const { getByTestId, findByText } = render(<LoginContainer />);
    const loginButton = getByTestId('login-button');

    fireEvent.click(loginButton);

    const errorMessage = await findByText(/username cannot be empty/i);
    expect(errorMessage).toBeInTheDocument();
});

  it('displays an error message when password is empty', async () => {
    const { getByTestId, findByText } = render(<LoginContainer />);
    const loginButton = getByTestId('login-button');

    fireEvent.click(loginButton);

    const errorMessage = await findByText(/password cannot be empty/i);
    expect(errorMessage).toBeInTheDocument();
  });

  it('calls the login function with the correct username and password', () => {
    const mockLogin = jest.fn();
    const { getByLabelText, getByTestId } = render(<LoginContainer login={mockLogin} />);
    const usernameInput = getByLabelText(/username/i);
    const passwordInput = getByLabelText(/password/i);
    const loginButton = getByTestId('login-button');

    fireEvent.change(usernameInput, { target: { value: 'testuser' } });
    fireEvent.change(passwordInput, { target: { value: 'testpass' } });
    fireEvent.click(loginButton);

    expect(mockLogin).toHaveBeenCalledWith('testuser', 'testpass');
  });
});