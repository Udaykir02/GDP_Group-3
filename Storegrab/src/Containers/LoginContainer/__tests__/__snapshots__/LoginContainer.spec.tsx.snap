// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingHorizontal": 20,
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/Assets/icons/bearcat_plaw.png",
                }
              }
              style={
                {
                  "alignSelf": "center",
                  "height": 220,
                  "justifyContent": "center",
                  "width": 200,
                }
              }
            />
          </View>
          <View
            style={
              {
                "flex": 1,
                "padding": 20,
                "width": "100%",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Email"
              style={
                {
                  "borderColor": "gray",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              testID="email-input"
              value=""
            />
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                {
                  "borderColor": "gray",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "marginBottom": 10,
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
              testID="password-input"
              value=""
            />
            <View
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "backgroundColor": "#68a0cf",
                  "borderColor": "#fff",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "marginLeft": 50,
                  "marginRight": 50,
                  "marginTop": 20,
                  "paddingBottom": 15,
                  "paddingTop": 15,
                }
              }
              testID="login-button"
            >
              <Text
                style={
                  [
                    {
                      "fontSize": 15,
                    },
                    {
                      "color": "#fff",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Login
              </Text>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  [
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Reset Password
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
